Week6/solution6.txt

```diff
@@ -0,0 +1,113 @@
+5   6       Om    80    80    56    67    56
+6   7      Sai    70    87    76    56    56
+7   8      Ram    60    45    86    76    78
+8   9     Shiv    34    67    78    66    45
+9  10  Shankar    56    10    56    78    34
+
+df.to_csv('c:\\sqlite3\\csv\\result.csv',index=False)
+
+df['TOTAL'],df['GRADE']=[[450,349,390,380,390,339,345,345,290,234],['A','B','B','B','B','C','C','C','D','D']]
+print(df)
+   ID     NAME  Sub1  Sub2  Sub3  Sub4  Sub5  TOTAL GRADE
+0   1     Heer    90    90    90    90    90    450     A
+1   2    Hanee    90    79    40    50    90    349     B
+2   3     Riya    90    90    30    90    90    390     B
+3   4    Kishu    90    60    50    90    90    380     B
+4   5     Hiya    90    30    90    90    90    390     B
+5   6       Om    80    80    56    67    56    339     C
+6   7      Sai    70    87    76    56    56    345     C
+7   8      Ram    60    45    86    76    78    345     C
+8   9     Shiv    34    67    78    66    45    290     D
+9  10  Shankar    56    10    56    78    34    234     D
+
+df.to_csv('c:\\sqlite3\\csv\\result.csv',index=False)
+
+num=df.nlargest(3,'TOTAL')
+print(num)
+   ID  NAME  Sub1  Sub2  Sub3  Sub4  Sub5  TOTAL GRADE
+0   1  Heer    90    90    90    90    90    450     A
+2   3  Riya    90    90    30    90    90    390     B
+4   5  Hiya    90    30    90    90    90    390     B
+
+
+
+
+//TEXT FILE PROGRAM 
+
+Program 1: Read Python.txt file and Print it in Reverse.
+import csv
+with open('c:\\sqlite3\\CSV\\python.txt','r',errors='ignore') as r:
+    reverse=r.read()
+    print('orginal string:',reverse)
+    print('reverse string:',reverse[::-1])
+
+
+Program 2: Read Python.txt file and Print total number of lines and words in it.
+import csv
+with open('c:\\sqlite3\\CSV\\python.txt','r') as r:
+    data=r.read()
+    print(data)
+with open('c:\\sqlite3\\CSV\\python.txt','r') as r:
+    l=r.readlines()
+    print('number of lines:',len(l))
+    words=data.split()
+print('Number of words in python file :', len(words))
+
+
+Program 3: Read Python.txt file and Print unique word with its count.
+import csv
+with open('c:\\sqlite3\\CSV\\python.txt','r') as rf:
+    file=rf.read()
+    print(file)
+    words=file.split()
+    u=set(words)
+    print("Unique Values Are:",u)
+    for word in u:
+        print(word, ":", words.count(word))
+
+
+Program 4: Read Python.txt file and Print largest and smallest word from it.
+import csv
+with open('c:\\sqlite3\\CSV\\python.txt','r',errors='ignore') as r:
+    reverse=r.read()
+    print(reverse)
+    f=reverse.split()
+    print("Largest String: ",max(f))
+    print("Smallest String: ",min(f))
+
+
+Program 5: Read Python.txt file. Convert it into upper case and write into another file "Python_Upper.txt"
+import csv
+with open('c:\\sqlite3\\CSV\\python.txt','r') as r:
+    data=r.read()
+    print(data)
+    w=data.upper()
+    print(w)
+    with open('c:\\sqlite3\\CSV\\python_upper.txt','w+') as w:
+        r.seek(0)

```

Comment:
What is the purpose of seek()..?

Reply:
seek() is used to change the position
seek(0) means the cursor is pointed at the start point

Reply:
Ok.. Good